lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -122,20 +197,68 @@ previewers.file_maker = function(filepath, bufnr, opts)
           end),
         })
       else
-        Path:new(filepath):_read_async(vim.schedule_wrap(function(data)
-          if not vim.api.nvim_buf_is_valid(bufnr) then
-            return
-          end
-          local ok = pcall(vim.api.nvim_buf_set_lines, bufnr, 0, -1, false, vim.split(data, "[\r]?\n"))
-          if not ok then
-            return
-          end
+        Job
+          :new({
+            command = "file",
+            args = { "--mime-type", "-b", filepath },
+            on_exit = function(j)
+              local mime_type = vim.split(j:result()[1], "/")[1]
+              if mime_type ~= "text" and type(ft) == "string" and ft == "" then
+                vim.schedule(function()
+                  set_timeout_message(bufnr, opts.winid, "Binary cannot be previewed")
+                end)
+                return
```

Comment:
I would only ask the `mime-type` oracle if ft is an empty string. Otherwise we kinda already know that its not a binary. I dont really know if its suitable to realize. We already are in callback hell here, even before your PR (so its not your fault)

Reply:
Makes sense. One thing I've realized, `file` is not a command on Windows and would introduce a dependency for any Windows user. From 10 minutes Google search, I'm not sure there is a cool workaround. So mime type detection only for Linux/Mac users? :sweat_smile:

Reply:
@Conni2461 

I've looked into mime type detection on Windows for too long and didn't find a satisfactory solution that introduces no dependencies (apart from maybe looking into the registry for the file extension which requires admin rights? :laughing: ). It would apparently easiest for Windows users to download `file` :man_shrugging: 

Do you then want to do mime-type detection for Linux/macOS users only or remove it entirely?

Reply:
we have mime type detection for `cat` since an eternity. We are doing this: https://github.com/nvim-telescope/telescope.nvim/blob/2c71ffec076dffab56333c0b66f95f0d38024cd2/lua/telescope/previewers/term_previewer.lua#L72-L78

not good because its sync, but we are only doing it if we have the dependency, otherwise we dont support it. In my opinion we dont actually have to support any platform that exists.

Reply:
I guess since it's `utils.get_os_command_output` doesn't work in this context (unless I'd await it, or something? Though would induce another callback I suppose). `capture` seems to work very well here. I can try to make it work nicely with `Job` if you prefer that though.